fairroot/base/sim/FairIon.cxx

```diff
@@ -23,7 +23,7 @@ FairIon::FairIon()
     , fMass(0)
 {}
 
-FairIon::FairIon(const char* name, Int_t z, Int_t a, Int_t q, Double_t e, Double_t mass)
+FairIon::FairIon(const std::string name, Int_t z, Int_t a, Int_t q, Double_t e, Double_t mass)
     : TNamed(name, "User defined ion")
```

Comment:
```suggestion
FairIon::FairIon(const TString& name, Int_t z, Int_t a, Int_t q, Double_t e, Double_t mass)
    : TNamed(name, "User defined ion")
```

When to use the string `by-value and move` pattern in ctors is discussed here: https://www.cppstories.com/2018/08/init-string-member/

Since we are forced by the `TNamed` ctor to create a temporary `TString` anyways, I guess,  there is nothing better to do here.

Reply:
[`TString::operator const char*`](https://root.cern.ch/doc/master/classTString.html#a58d6744a8da98ac00c160809a15645ba) :((( - how can it be that ROOT uses every anti-pattern that is possible...

Reply:
You could disambiguate this maybe?
```cpp
TNamed(name, TString{"User defined ion"})
```

Reply:
Personal note: I don't like the proliferation of TString into our public APIs. Especially when we can stay with more generic ones like `const char*` (or `czstring`–yes, please let's discuss somewhere else)

If https://github.com/root-project/root/issues/15434 comes into reality, we could consider `string_view`.